### PR TITLE
Initial talon language support

### PR DESCRIPTION
**Pull request by phillco**
Resolves https://github.com/cursorless-dev/cursorless/issues/817

**Condition**
- [x] context match
- [x] every context match

**Command**
- [x] command
- [x] every command

**Statement**
- [x] command line
- [x] command block
- [x] settings block
- [x] script action call
- [x] script key action call
- [x] script sleep action call
- [x] script assignment
- [ ] every statement in file
- [x] every statement in command block
- [x] every statement in settings block

**Name / Collection key**
- [x] context match left
- [x] command left
- [x] settings block left
- [x] setting left
- [x] tag() left
- [x] key() left
- [x] gamepad() left
- [x] face() left
- [x] parrot() left
- [x] every name/key in context block
- [x] every name/key in file

**Value** 
- [x] context match right
- [x] command right
- [x] settings block right
- [x] setting right
- [x] tag() right
- [x] key() right
- [x] gamepad() right
- [x] face() right
- [x] parrot() right
- [x] every value in context block
- [x] every value in settings block
- [x] every value in file

**Interior**
- [x] command right
- [ ] settings block right
- [x] every command interior in file

**Function call**
- [x] action
- [x] key action
- [x] sleep action
- [x] every call in command block

**Function callee**
- [x] action
- [ ] key action
- [ ] sleep action
- [ ] every callee in command block

**Argument or parameter**
- [x] action
- [x] key action
- [x] sleep action
- [x] every argument in action

**Miscellaneous**
- [ ] Update Tree sitter `implicit_string`

## Checklist

- [x] I have added [tests](https://www.cursorless.org/docs/contributing/test-case-recorder/)
- [ ] I have updated the [docs](https://github.com/cursorless-dev/cursorless/tree/main/docs) and [cheatsheet](https://github.com/cursorless-dev/cursorless/tree/main/cursorless-talon/src/cheatsheet)
- [ ] I have not broken the cheatsheet

### CODE REVIEW

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearEveryValue2.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    -
+    aaa: bbb
+  selections:
+    - anchor: {line: 0, character: 0}
+      active: {line: 0, character: 0}
+  marks: {}
+finalState:
+  documentContents: |-
+    tag:
+    tag:
```
Comment (pokey) [inline packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearEveryValue2.yml:25]: These lines should have spaces at the end; see https://github.com/wenkokke/tree-sitter-talon/issues/22

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearKey3.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    -
+    aaa: bbb
+  selections:
+    - anchor: {line: 1, character: 21}
+      active: {line: 1, character: 21}
+  marks: {}
+finalState:
+  documentContents: |-
+    tag: user.aaa
+    and not : user.bbb
```
Comment (pokey) [inline packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearKey3.yml:25]: Question: is this what we want? Not necessarily what I expected, but could be useful, and there is a logic to it. @AndreasArvidsson suggests that maybe "name" should do this and "key" should be everything left of the `:`

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearValue2.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    tag: user.aaa
+    -
+    aaa: bbb
+  selections:
+    - anchor: {line: 0, character: 0}
+      active: {line: 0, character: 0}
+  marks: {}
+finalState:
+  documentContents: |-
+    tag:
```
Comment (pokey) [inline packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearValue2.yml:23]: As above, this line should have a space at the end; see https://github.com/wenkokke/tree-sitter-talon/issues/22

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearValue3.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    -
+    aaa: bbb
+  selections:
+    - anchor: {line: 1, character: 0}
+      active: {line: 1, character: 0}
+  marks: {}
+finalState:
+  documentContents: |-
+    tag: user.aaa
+    and not tag:
```
Comment (pokey) [inline packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearValue3.yml:25]: As above re space / https://github.com/wenkokke/tree-sitter-talon/issues/22

File: packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearEveryValue2.yml (hunk 1 above)
Comment (AndreasArvidsson) [reply packages/cursorless-vscode-e2e/src/suite/fixtures/recorded/languages/talon/clearEveryValue2.yml:25]: PR created
https://github.com/wenkokke/tree-sitter-talon/pull/23

**Comment by phillco** [comment]
@AndreasArvidsson given that you did all the work here, I feel the final commit should probably be attributed to you or pokey. If we squash merge this it'll be attributed to me. :) 

We can achieve that by closing this, making a new branch, and having you open a new PR under your name

**Comment by AndreasArvidsson** [comment]
Is that the simplest way? @pokey do we have any control over who gets contributed when we do a merge?

**Comment by pokey** [comment]
We'll all show up as co-authors, but yeah the only way for you to be the primary author is to do what @phillco suggests. Given there's no activity yet on this PR, might make sense to do that. @phillco and I didn't do much here so I think it's safe to call this your PR. Or if you don't care that's fine too. But lmk either way because I have some comments I'd like to leave so it would be good to leave them on the final PR 😊

**Comment by AndreasArvidsson** [comment]
Okay I will close this and make a new one then :)

**Comment by AndreasArvidsson** [comment]
Replaced by #1555

**Pull request closed without merging**